### DISCUSSION TITLE
Change Weather units

### DISCUSSION THREAD

**Discussion by hamiltro84** [MMRIZE/MMM-CalendarExt3#48] [Q&A]
I have the weather module all configured, and it shows imperial units, but when I show the weather information on the calendar it all seems to be in metric. I have defined units within the MMM-CalendarExt3 config module. I have setup my locate to:
`locale: 'en-US-u-ca-iso8601-hc-h23'`

I will be honest I would not be surprised if the key to unlocking the imperial units is hidden in the locale code. Maybe an easier way to  convert time to 23h or just drop the AM/PM to save some room on the calendar.  Sorry this is about weather units not time.

**Comment by eouia** [comment]
The weather information is derived from default `weather` module(or 3rd party module that emit equivalent notification), so the value respects the setting of the original weather module. When you set the `weather` module as imperial, the value would be Fahrenheit. It is not decided by this module.

**Reply by hamiltro84** [reply]
even with the weather module set to imperial the MMM-CalendarEXT3 does not show the imperial unit but rather the default metric.

Here is a screen shot of the Weather Module and MMM-CalendarEXT3 running

![Screen Shot 2023-01-14 at 10 01 24 PM](https://user-images.githubusercontent.com/24468682/212521620-8c5fbcfa-eaf2-412a-ba54-db7bc2353824.png)

**Comment by eouia** [comment]
Hmmm... This is very weird. Can I see your config of the weather module (without your private key)? Because there is no conversion logic between imperial/metric in this module. This module only shows the data from the weather module directly.

**Reply by hamiltro84** [reply]
Sure, here you go, also included the MMM-CalendarExt3 incase I has something set wrong messing it up. 
```
{
	module: "weather",
	position: "top_right",
	config: {
		weatherProvider: "openweathermap",
		type: "forecast",
		units: "imperial",
		tempUnits: "imperial",
		roundTemp: true,
		updateInterval: 3600000,
		colored: true,
		fade: false,
		maxNumberOfDays: "7",
		weatherEndpoint: "/forecast",
		locationID: "4515843",
		location: 'Kettering',
		apiKey: "Secret_apiKey"
	}
},
{
	module: "MMM-CalendarExt3",
	position: "middle",
	title: "",
	config: {
		mode: 'week',
		weekIndex: 0,
		weeksInView: 4,
		instanceId: 'basicCalendar',
		useWeather: true,
		weatherLocationName:'Kettering',
		locale: 'en-US-u-ca-iso8601-hc-h23',
		maxEventLines: 5,
		firstDayOfWeek: 0,
		calendarSet: ['us_holiday'],
	}
},
```

**Reply by eouia** [reply]
To test more detail, Can you share the LocationID and Location also?

**Reply by hamiltro84** [reply]
I updated my code to include everything, only thing missing is my Secret apiKey

**Comment by eouia** [comment]
<img width="294" alt="image" src="https://user-images.githubusercontent.com/1720610/214587378-dcdc5ca0-0a50-43bc-a335-fd8a727663e5.png">
Hmmm... With my test, the unit of temperature is coming from the weather module directly.

**Reply by hamiltro84** [reply]
I tasted on a fresh deploy on my RPI4 but had the same issue. What do you have listed in your CSS to get the temperate to display. this could be my issue. 
mine as a follows
```
.CX3 .cellHeader .cellWeather .temperature {
  display: inline-block;
}
.CX3 .cellHeader .cellWeather .temperature {
	color: var(--defaultcolor);
}
```

**Reply by eouia** [reply]
Nothing different with yours.
Ok, I'll provide you some test version in a time.

**Comment by eouia** [comment]
I uploaded `temp_test` version on the repository. To apply it;
```sh
// GOTO MMM-CalendarExt3 directory

git pull
git checkout temp_test
```

Then go to MM directory
```sh
npm run start:dev
```
>> Warning: You have to terminate your current MM before this. If you are using PM2, stop it.

You can see `dev-console` somewhere on the MM screen (usually right side)


![image](https://user-images.githubusercontent.com/1720610/216918241-cac2b0d4-cf71-4ef7-841e-16b46734f740.png)

I modified the codes to spit out some more information there.

<img width="257" alt="image" src="https://user-images.githubusercontent.com/1720610/216918442-a818fb98-b21f-4f9c-a1ea-ffa0bcca21cc.png">

Check `ORIGINAL WEATHER` array. Confirm `maxTemperature`, `minTemperature`, `units`, `tempUnits`.

**Reply by hamiltro84** [reply]
Still having issues, I am just going to start from scratch, even scratch configurations. Thanks for the help so far, I will let you know what happens with the true reset button
![image](https://user-images.githubusercontent.com/24468682/221377381-838abd50-959c-44bd-99db-f5bacd3ff8b4.png)

**Comment by eouia** [comment]
As you can see the dev-console, original weather information from API itself seems set as metric unit (9.5 and 2.72 : max/minTemperature) So nothing wrong on my module.

But I don’t know why the weather module could show Farhenheit with Celcius data.

**Comment by KristjanESPERANTO** [comment]
This was fixed in MagicMirror² core with [PR #3519](https://github.com/MagicMirrorOrg/MagicMirror/pull/3519) (released in v2.29.0, October 2024) :slightly_smiling_face:
